apps/user/serializers.py

```diff
@@ -222,3 +227,28 @@ class EntryCommentUserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
         fields = ('id', 'name', 'email', 'organization', 'display_picture_url',)
+
+
+class PasswordChangeSerializer(serializers.ModelSerializer):
+    old_password = serializers.CharField(required=True, write_only=True)
+    new_password = serializers.CharField(required=True, write_only=True)
+
+    class Meta:
+        model = User
+        fields = ['old_password', 'new_password']
+
+    def validate_old_password(self, password):
+        user = self.context['request'].user
+        if not user.check_password(password):
+            raise serializers.ValidationError('Invalid Old Password')
+        return password
+
+    def validate_new_password(self, password):
+        validate_password(password)
+        return password
+
+    def save(self, **kwargs):
+        user = self.context['request'].user
+        user.set_password(self.validated_data['new_password'])
+        user.save()
+        send_password_change(user)
```

Comment:
Let's do 3 things here.
- Use transaction.on_commit
- change send_password_change -> send_password_changed_notification
- Make send_password_changed_notification celery task and use that.

```python3
@shared_task
def send_password_changed_notification(user_id):
    .....
....
transaction.on_commit(
    lambda: send_password_changed_notification.delay(user.id)
)
```